inference-engine/src/transformations/src/transformations/common_optimizations/broadcast_elementwise_fusion.cpp

```diff
@@ -0,0 +1,70 @@
+// Copyright (C) 2018-2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/common_optimizations/broadcast_elementwise_fusion.hpp"
+
+#include <ngraph/opsets/opset5.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::BroadcastElementwiseFusion, "BroadcastElementwiseFusion", 0);
+
+bool is_eliminate_broadcast(const ngraph::PartialShape & input_shape, const ngraph::PartialShape & broadcast_shape) {
+    if (input_shape.rank().is_dynamic() || broadcast_shape.rank().is_dynamic()) {
+        return false;
+    }
+
+    const int64_t & input_shape_rank = input_shape.rank().get_length();
+    const int64_t & broadcast_shape_rank = broadcast_shape.rank().get_length();
+    if (broadcast_shape_rank > input_shape_rank) {
+        //We can not eliminate broadcast op because
+        //in the case input_shape will be broadcast
+        return false;
+    }
+    for (int64_t i_dim = input_shape_rank - 1, b_dim = broadcast_shape_rank - 1; i_dim >= 0 && b_dim >=0; --i_dim, --b_dim) {
+        if (input_shape[i_dim].is_static() && broadcast_shape[b_dim].is_static()) {
+            const auto &input_shape_dim = input_shape[i_dim].get_length();
+            const auto &broadcast_shape_dim = broadcast_shape[b_dim].get_length();
+            if (input_shape_dim != broadcast_shape_dim && broadcast_shape_dim != 1) {
+                //We can not eliminate broadcast op because
+                //input_shape will be broadcast
+                return false;
+            }
+        } else if (input_shape[i_dim].is_dynamic() && broadcast_shape[i_dim].is_static() &&
+                   broadcast_shape[i_dim].get_length() != 1) {
+            return false;
+        }
```

Comment:
I think that the other case is missed here when we cannot eliminate `Broadcast`: `broadcast_shape[i_dim].is_dynamic()` - we are unable to say anything about this dimension.

Reply:
Please add test to cover this case.

Reply:
[2,1,4] [2,?,4] False